### ISSUE TITLE
Dexter UI config causes NullReferenceException

### ISSUE THREAD

**Issue by jscottnz** [Appdynamics/AppDynamics.DEXTER#54]
**Describe the bug**

When no comparison application details are provided in the UI the generated config results in:
`Input.ConfigurationComparisonReferenceCriteria : {}`

This causes a `NullReferenceException` here https://github.com/Appdynamics/AppDynamics.DEXTER/blob/3b1960c5c89c424d79f53921a6d81a92cd0c93aa/Program.cs#L267

Replacing with `Input.ConfigurationComparisonReferenceCriteria : null` fixes the issue.

➜  dexter core-osx/AppDynamics.Dexter --jobfile xx.json
AppDynamics DEXTER Version 1.2.37.0
Checking input job file xx.json
Checking output folder AppD.Dexter.Out
Creating job output folder xx/config.dexter
Processing input job file to output job file
System.NullReferenceException: Object reference not set to an instance of an object.
   at AppDynamics.Dexter.Program.Main(String[] args) in C:\appdynamics\AppDynamics.DEXTER\src\Program.cs:line 267
Application execution took 00:00:01.4525810 (1452 ms)

**To Reproduce**

1. Go to Dexter UI
2. Create configuration
3. In the controller comparison, delete the entries

**Expected behavior**
There is no NRE.

**Comment by danielodievich** [comment]
Why are you deleting ConfigurationComparisonCriteria? It is expected to be present

**Comment by danielodievich** [comment]
Oh, I see what is going on. I have changed the code in the following way:

```
                    // Validate Configuration Comparison selection
                    if (jobConfiguration.Input.ConfigurationComparisonReferenceCriteria == null)
                    {
                        jobConfiguration.Input.ConfigurationComparisonReferenceCriteria = new JobTarget();
                        jobConfiguration.Input.ConfigurationComparisonReferenceCriteria.Controller = JobStepBase.BLANK_APPLICATION_CONTROLLER;
                        jobConfiguration.Input.ConfigurationComparisonReferenceCriteria.Application = JobStepBase.BLANK_APPLICATION_APPLICATION;
                    }
                    else
                    {
                        if (jobConfiguration.Input.ConfigurationComparisonReferenceCriteria.Controller == null ||
                            jobConfiguration.Input.ConfigurationComparisonReferenceCriteria.Application == null)
                        {
                            jobConfiguration.Input.ConfigurationComparisonReferenceCriteria = new JobTarget();
                            jobConfiguration.Input.ConfigurationComparisonReferenceCriteria.Controller = JobStepBase.BLANK_APPLICATION_CONTROLLER;
                            jobConfiguration.Input.ConfigurationComparisonReferenceCriteria.Application = JobStepBase.BLANK_APPLICATION_APPLICATION;
                        }
                        else
                        {
                            jobConfiguration.Input.ConfigurationComparisonReferenceCriteria.Controller = jobConfiguration.Input.ConfigurationComparisonReferenceCriteria.Controller.TrimEnd('/');
                        }
                    }
```

Should work, no?

**Comment by danielodievich** [comment]
Fixed by 37777e24053970e54cff74e8ddd99bce00cd0258
